FirebaseStorageUITests/FUIImageViewCategoryTests.m

```diff
@@ -0,0 +1,188 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+@import XCTest;
+@import FirebaseStorageUI;
+
+#import "UIImageView+FirebaseStorage.h"
+
+@class MockDownloadTask, MockStorageReference, MockCache;
+
+@interface MockDownloadTask : NSObject <FUIDownloadTask>
+@property (nonatomic, readwrite, getter=isCancelled) BOOL cancelled;
+@end
+
+@implementation MockDownloadTask
+
+- (void)cancel {
+  self.cancelled = YES;
+}
+
+@end
+
+
+@interface MockCache : NSObject <FUIImageCache>
+@property (nonatomic, readonly) NSMutableDictionary *cached;
+@end
+
+@implementation MockCache
+
+- (instancetype)init {
+  self = [super init];
+  if (self != nil) {
+    _cached = [NSMutableDictionary dictionary];
+  }
+  return self;
+}
+
+- (void)storeImage:(UIImage *)image forKey:(NSString *)key {
+  self.cached[key] = image;
+}
+
+- (UIImage *)imageFromDiskCacheForKey:(NSString *)key {
+  return self.cached[key];
+}
+
+- (UIImage *)imageFromMemoryCacheForKey:(NSString *)key {
+  return self.cached[key];
+}
+
+@end
+
+
+@interface MockStorageReference : NSObject <FUIStorageReference>
+@property (nonatomic, readwrite) NSString *fullPath;
+@end
+
+@implementation MockStorageReference
```

Comment:
I'll do this one for you, since it's worth comparison to OCMock:
```
// Always use
id mockStorageReference = OCMClassMock([FIRStorageReference class]);

// Test failure case
NSError *error = [NSError errorWithDomain:@"FUITestDownloadErrorDomain" code:1 userInfo:nil];
void (^errorCallback)(NSData *, NSError *) = [OCMArg invokeBlockWithArgs:[NSNull null], error, nil];
OCMStub([mockStorageReference dataWithMaxSize:[OCMArg any] callback:errorCallback])

// Test success case
void (^successCallback)(NSData *, NSError *) = [OCMArg invokeBlockWithArgs:[NSData new], [NSNull null], nil];
OCMStub([mockStorageReference dataWithMaxSize:[OCMArg any] callback:errorCallback])
```